src/cloud-foundry-api-client.js

```diff
@@ -19,6 +19,48 @@ class CloudFoundryAPIClient {
       )).then(body => this._filterAppsResponse(JSON.parse(body)));
   }
 
+  fetchAppStats(appGUID) {
+    return this._authClient.accessToken().then(token => this._request(
+        'GET',
+        `/v2/apps/${appGUID}/stats`,
+        token
+      ));
+  }
+
+  fetchAppInstanceStates(container) {
+    return this.fetchAppStats(container.guid)
+      .then(stats =>
+        ({
+          guid: container.guid,
+          name: container.name,
+          states: this._appInstanceStates(JSON.parse(stats)),
+        })
+      );
+  }
+
+  fetchAllAppInstanceErrors(buildContainers) {
+    const instanceErrors = [];
+    const promises = [];
+    let states;
+    let i;
+    for (i = 0; i < buildContainers.length; i += 1) {
+      promises.push(this.fetchAppInstanceStates(buildContainers[i]));
+    }
+
+    return Promise.all(promises)
+    .then((instanceStates) => {
+      for (i = 0; i < instanceStates.length; i += 1) {
```

Comment:
Again here, we could probably clean this up using `Array.forEach` or `Array.map`.